### ISSUE TITLE
[BUG] Doesn't work on macOs 10.13.6 & 10.15.2

### ISSUE THREAD

**Issue by franzuu** [WoodyLetsCode/WLED-GUI#4]
**Describe the bug**
Hi! 
I have two versions of macOs (two computers). When I try to open the app, it shows error. Is this app compiled only for some specific versions of macOs?

**Expected behavior**
The app should open?!

**Screenshots**
High Sierra (10.13.6) gives this error:
![Screen Shot 2020-10-16 at 10 31 43](https://user-images.githubusercontent.com/18600583/96226739-c1c20e80-0f9b-11eb-9ceb-83c50f82fb74.png)

Catalina (10.15.2) gives this error:
![0-02-05-fbe5aced3fb6166b7fd06e76908899608963ffa84917e514c8369712ffe6e70f_1c6d9e18ad9f6b](https://user-images.githubusercontent.com/18600583/96226833-e1f1cd80-0f9b-11eb-8ecf-5f90a9bffd0e.jpg)

**System:**
 - OS: macOs
 - OS Versions: 10.13.6 & 10.15.2
 - App version: 0.4.1

**Comment by NeariX67** [comment]
Same on 10.15.6

**Comment by WoodyLetsCode** [comment]
Hi Guys!
Thank you for telling me this! 
The problem is I don't have a computer with MacOS, so i couldn't test it before uploading. I assumed that it will work, but apparently it won't. 😟
Now i have a question:
Is this a new bug, or did it work in previous version of WLED-GUI?
If this is your first time using WLED-GUI please tell me. You don't have to test every previous version. 🙂

**Comment by franzuu** [comment]
Thank you for quick reply! For me, it is first time using it.

**Comment by NeariX67** [comment]
Running a terminal command to launch the GUI returns this:
`justin@Justins-iMac WLED-darwin-x64 % open -a WLED.app 
LSOpenURLsWithRole() failed for the application /Users/justin/Downloads/WLED-darwin-x64/WLED.app with error -10810.`
Can't get any other error out of it

**Comment by NeariX67** [comment]
Running
`chmod +x WLED.app/Contents/MacOS/WLED`
makes the app executable. Then you can allow the launch in the system preferences privacy tab.
This brings up a new messages that apple can't trust the developer... After clicking open anyway [...]

As I'm writing this comment, it somehow doesn't work anymore and can't get the error to paste here... Damn...

**Comment by WoodyLetsCode** [comment]
@franzuu Did you run the` chmod +x WLED.app` command?

**Comment by franzuu** [comment]
Tried chmod, still the same on both computers

**Comment by WoodyLetsCode** [comment]
Hey,
sorry for the late reply.
Have you tried changing the permission to 755 for the folder? What was the permission before changing? And wich user is the owner of the folder?

**Comment by Walkersneps** [comment]
Hi! I was getting the same problem and I just managed to solve it while writing this.

The error popup gave me the following error:

```
Process:               WLED [956]
Path:                  /Volumes/VOLUME/*/WLED.app/Contents/MacOS/WLED
Identifier:            com.electron.wled
Version:               0.4.0 (0.4.0)
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           WLED [956]
User ID:               501

Date/Time:             2020-11-27 13:29:39.857 +0100
OS Version:            Mac OS X 10.13.6 (17G14042)
Report Version:        12
Anonymous UUID:        1DE3E31C-1BBE-B380-6772-07A430E39321


Time Awake Since Boot: 3800 seconds

System Integrity Protection: enabled

Notes:                 Translocated Process

Crashed Thread:        0

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    DYLD, [0x1] Library missing

Application Specific Information:
dyld: launch, loading dependent libraries

Dyld Error Message:
  Library not loaded: @rpath/Electron Framework.framework/Electron Framework
  Referenced from: /Volumes/VOLUME/*/WLED.app/Contents/MacOS/WLED
  Reason: no suitable image found.  Did find:
	/private/var/folders/mt/gk57y59n4xg5w5t2jvm7fz640000gn/T/AppTranslocation/4142FACC-3EDB-4297-A5D5-FD6742068662/d/WLED.app/Contents/MacOS/../Frameworks/Electron Framework.framework/Electron Framework: file too short
	/private/var/folders/mt/gk57y59n4xg5w5t2jvm7fz640000gn/T/AppTranslocation/4142FACC-3EDB-4297-A5D5-FD6742068662/d/WLED.app/Contents/MacOS/../Frameworks/Electron Framework.framework/Electron Framework: stat() failed with errno=1

Binary Images:
       0x102e65000 -        0x102e8cff3 +com.electron.wled (0.4.0 - 0.4.0) <5CF444E6-280A-33A3-87FF-B56A6B4EC25C> /var/folders/*/WLED.app/Contents/MacOS/WLED
       0x10e2f1000 -        0x10e33badf  dyld (551.5) <CB9BFB56-4511-36F1-A546-891FF770C01C> /usr/lib/dyld
```

From `Library not loaded: @rpath/Electron Framework.framework/Electron Framework` I got that for some reason it couldn't load the framework, so I worked it around by copying all the files contained in `WLED.app/Contents/Frameworks/Electron\ Framework.framework/Versions/A/` into `WLED.app/Contents/Frameworks/Electron\ Framework.framework/`. Then proceeded to the the same with the other 3 frameworks (`Squirrel`, `ReactiveCocoa`, `Mantle`).

**Comment by WoodyLetsCode** [comment]
@Walkersneps Thank you very much for telling me this!
Could you please share how you had unarchived the zip?

**Comment by Walkersneps** [comment]
Sure! I just let it use the default macOS Archive Utility.
Do you think the unzipping program may be the problem?

**Comment by WoodyLetsCode** [comment]
I read in another article that some useres had problems with "iZip". But you don't use it, so it is not the problem.
I assume there is a problem with the links. But I have to take a closer look.

**Comment by WoodyLetsCode** [comment]
@Walkersneps Just to make it clear, [this version](http://www.mediafire.com/file/rqy02azr1ajaous/WLED-darwin-x64.zip/file) works for you?
I copied everything like you told me.

**Comment by Walkersneps** [comment]
@WoodyLetsCode That's a bit weird: Archive Utility doesn't unzip it, giving `(Error 21 - Is a directory.)`, whereas trying with "The Unarchiver" throws some error popups but after ignoring them, I get a correctly unzipped directory with a working app.

**Comment by WoodyLetsCode** [comment]
@Walkersneps Can you try [this](https://github.com/srg74/WLED-wemos-shield/blob/master/resources/WLED_GUI_for_mac/WLED-0.4.1.dmg) one? This should work.

**Comment by Walkersneps** [comment]
@WoodyLetsCode Yep, this one works fine

**Comment by WoodyLetsCode** [comment]
Thank you for telling me this!

**Comment by WoodyLetsCode** [comment]
Hello @franzuu , @NeariX67 and @Walkersneps ,
thank you for reporting me this. 
Just a few minutes ago i published [version 0.5.0](https://github.com/WoodyLetsCode/WLED-GUI/releases/tag/v0.5.0).
In this version you have a .dmg-file which should work for you.
Have a good time!
